src/main/java/com/mitchej123/hodgepodge/core/util/AnchorAlarm.java

```diff
@@ -0,0 +1,101 @@
+package com.mitchej123.hodgepodge.core.util;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.world.WorldServer;
+
+import net.minecraftforge.common.DimensionManager;
+
+import com.mitchej123.hodgepodge.Hodgepodge;
+import com.mojang.authlib.GameProfile;
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import cpw.mods.fml.common.gameevent.PlayerEvent;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import mods.railcraft.common.blocks.machine.alpha.TileAnchorWorld;
+import mods.railcraft.common.plugins.forge.PlayerPlugin;
+
+public class AnchorAlarm {
+    private static final String NBT_KEY = "GT_RC_AnchorAlarmList";
+    public static void addNewAnchor(EntityLivingBase entityliving, TileEntity te) {
+        if (entityliving instanceof EntityPlayerMP) {
+            GameProfile owner = ((EntityPlayerMP)entityliving).getGameProfile();
+            for(Object obj : te.getWorldObj().playerEntities) {
+                if (obj instanceof EntityPlayerMP) {
+                    EntityPlayerMP player = (EntityPlayerMP) obj;
```

Comment:
Maybe I'm being stupid here, but shouldn't `entityliving == player` anyway? Why do you need to loop through all players to find the same player?

Reply:
Huh, indeed, that code was in the different part, where I didn't know the player.